packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -841,6 +845,37 @@ export class HardhatNode extends EventEmitter {
     await this._txPool.updatePendingAndQueued();
   }
 
+  public async traceTransaction(hash: Buffer): Promise<object> {
+    const block = await this.getBlockByTransactionHash(hash);
+    if (block === undefined) {
+      throw new InvalidInputError(
+        `Unable to find a block containing transaction ${bufferToHex(hash)}`
+      );
+    }
+
+    return this._runInBlockContext(
+      new BN(block.header.number).subn(1),
+      async () => {
+        const blockGasLimit = this.getBlockGasLimit();
+        const gasLeft = blockGasLimit.clone();
+        for (const tx of block.transactions) {
+          const txHash = tx.hash();
+          if (txHash.equals(hash)) {
+            this._vmDebugTracer.enableTracing();
```

Comment:
We discussed offline with @fvictorio that the enable/disable should probably be in a try/finally clause, in case the tracer throws in the future.